File: TopCoder/src/main/java/com/leetcode/LongestConsecutive.py

from typing import List


class Solution:

    def longestConsecutive(self, nums: List[int]) -> int:

        nums_set = set(nums)
        max_value = 0
        for num in nums:
            if num-1 not in nums_set:
                next = num
                while next in nums_set:
                    next = next + 1

                max_value = max(max_value, (next - num))

        return max_value

if __name__ == '__main__':
    sol = Solution()
    res = sol.longestConsecutive([100, 4, 200, 1, 3, 2])
    print(res)

File: TopCoder/src/main/java/com/leetcode/LongestConsecutive.java
package com.leetcode;

import java.util.Arrays;
import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

public class LongestConsecutive {

    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int i = 0 ; i < nums.length ; i++) {
            set.add(nums[i]);
        }
        int max = Integer.MIN_VALUE;
        for (int i = 0 ; i < nums.length ; i++)  {
            if (set.contains(nums[i])) {
                int next = nums[i];
                while (set.contains(next)) {
                    next = next + 1;
                }
                max = Math.max(max, (next) - nums[i]);
            }
        }
        return max == Integer.MIN_VALUE ? 0 : max;
    }

    public static void main(String[] args) {
        int[] nums = new int[]{100, 4, 200, 1, 3, 2};
        int res = new LongestConsecutive().longestConsecutive(nums);
        System.out.println(res);
    }
}
